File: Stage15/Python/Problem1735.py
import math

def main():
    A, B = map(int, input().split())
    C, D = map(int, input().split())
    
    numerator = A * D + B * C
    denominator = B * D

    g = math.gcd(numerator, denominator)
    numerator //= g
    denominator //= g
    
    print(numerator, denominator)
    
if __name__ == "__main__":
    main()

File: Stage15/C/Problem1735.c
#include <stdio.h>

int gcd(int a, int b) {
    while (b != 0) {
        int temp = a % b;
        a = b;
        b = temp;
    }
    return a;
}

int main(void) {
    int A, B, C, D;
    scanf("%d %d", &A, &B);
    scanf("%d %d", &C, &D);

    int numerator = A * D + B * C;
    int denominator = B * D;

    int g = gcd(numerator, denominator);
    numerator /= g;
    denominator /= g;

    printf("%d %d\n", numerator, denominator);
    return 0;
}